File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/ReadPropertyValuesResponse.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.DataValue;
import org.eclipse.milo.opcua.stack.core.types.builtin.StatusCode;
import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
import org.opcfoundation.webserver.types.common.UaChildId;

import java.util.HashMap;
import java.util.Map;

public class ReadPropertyValuesResponse {
    private final Map<UaChildId, DataValue> results;

    public ReadPropertyValuesResponse()
    {
        results = new HashMap<>();
    }

    public void setValue(String elementName, Variant value)
    {
        results.put(new UaChildId(elementName), new DataValue(value, StatusCode.GOOD, null, null));
    }

    public void setError(String elementName, StatusCode errorCode)
    {
        results.put(new UaChildId(elementName), new DataValue(Variant.NULL_VALUE, errorCode, null, null));
    }

    public Map<UaChildId, DataValue> getResults() {
        return results;
    }
}

File: opcua-webservice-node/source/src/server/service/message/digitialtwin/ReadPropertyValuesResponse.ts
import { StatusCodes } from "opcua-webapi";
import { makeUaStatusCode, UaDataValue, UaStatusCode, UaVariant } from "opcua-webapi-ts";
import { UaChildId } from "../../../types/common/UaChildId";

export class ReadPropertyValuesResponse {
    private readonly _results: Map<string, UaDataValue>;

    constructor(results: Map<string, UaDataValue> = new Map()) {
        this._results = results;
    }

    setValue(elementName: string | UaChildId, value: UaVariant): void {
        this._results.set(
            this.toKey(elementName),
            new UaDataValue(value, makeUaStatusCode(StatusCodes.Good)),
        );
    }

    setError(elementName: string | UaChildId, errorCode: UaStatusCode): void {
        this._results.set(
            this.toKey(elementName),
            new UaDataValue(UaVariant.null(), errorCode),
        );
    }

    get results(): Map<string, UaDataValue> {
        return this._results;
    }

    private toKey(elementName: string | UaChildId): string {
        if (typeof elementName === "string") {
            return elementName;
        }

        return elementName.toString();
    }
}